contract/lst-oracle/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+#![allow(clippy::needless_pass_by_value)]
+
+use near_sdk::{
+    assert_one_yocto, borsh::BorshSerialize, collections::UnorderedMap, env, near,
+    serde::de::DeserializeOwned, serde_json, AccountId, BorshStorageKey, Gas, IntoStorageKey,
+    PanicOnDefault, Promise, PromiseResult,
+};
+use near_sdk_contract_tools::{owner::Owner, Owner};
+use templar_common::{
+    define_list,
+    number::Decimal,
+    oracle::{
+        price_transformer::PriceTransformer,
+        pyth::{ext_pyth, OracleResponse, PriceIdentifier},
+    },
+    self_ext,
+};
+
+#[derive(BorshSerialize, BorshStorageKey)]
+#[borsh(crate = "near_sdk::borsh")]
+enum StorageKey {
+    Transformers,
+}
+
+#[derive(Debug, Owner, PanicOnDefault)]
+#[near(contract_state)]
+pub struct Contract {
+    pub oracle_id: AccountId,
+    pub transformers: UnorderedMap<PriceIdentifier, PriceTransformer>,
+}
+
+#[near]
+impl Contract {
+    #[init]
+    pub fn new(oracle_id: AccountId) -> Self {
+        let mut self_ = Self {
+            oracle_id,
+            transformers: UnorderedMap::new(StorageKey::Transformers.into_storage_key()),
+        };
+
+        Owner::init(&mut self_, &env::predecessor_account_id());
+
+        self_
+    }
+
+    pub fn get_oracle_id(&self) -> AccountId {
+        self.oracle_id.clone()
+    }
+
+    define_list! {
+        pub fn list_transformers(&self) -> Vec<PriceIdentifier> {
+            self.transformers.keys()
+        }
+    }
+
+    pub fn get_transformer(&self, price_identifier: PriceIdentifier) -> Option<PriceTransformer> {
+        self.transformers.get(&price_identifier)
+    }
+
+    #[payable]
+    pub fn create_transformer(&mut self, price_id: PriceIdentifier, entry: PriceTransformer) {
+        assert_one_yocto();
+        self.assert_owner();
+
+        if self.transformers.insert(&price_id, &entry).is_some() {
+            env::panic_str("Price identifier collision");
+        }
+    }
+
+    // impl Pyth:
+
+    pub fn price_feed_exists(&self, price_identifier: PriceIdentifier) -> bool {
+        self.transformers.get(&price_identifier).is_some()
+    }
+
+    pub fn list_ema_prices_no_older_than(
+        &self,
+        price_ids: Vec<PriceIdentifier>,
```

Comment:
Early exit on this `Vec` would be safe, unless its always expected to have `PriceIdentifier`s.